lib/checkcondition.cpp

```diff
@@ -707,7 +707,8 @@ void CheckCondition::multiCondition2()
                                 return ChildrenToVisit::op1_and_op2;
 
                             if ((!cond1->hasKnownIntValue() || !secondCondition->hasKnownIntValue()) &&
-                                isSameExpression(mTokenizer->isCPP(), true, cond1, secondCondition, mSettings->library, true, true, &errorPath)) {
+                                isSameExpression(mTokenizer->isCPP(), true, cond1, secondCondition, mSettings->library, true, true, &errorPath) &&
+                                !isExpressionChangedAt(cond1, secondCondition, 0, false, mSettings, mTokenizer->isCPP())) {
```

Comment:
Dont you need to check if the expression is changed before the `secondCondtion` not at the `secondCondition`?

Above at line 663, we check if the variables are changed in the second condition. It should probably just be changed to just use `isExpressionChanged(cond1, condStartToken, condEndToken, 0, false, mSettings, mTokenizer->isCPP())`.